packages/caver-contract/src/index.d.ts

```diff
@@ -0,0 +1,141 @@
+/*
+    Copyright 2021 The caver-js Authors
+    This file is part of the caver-js library.
+
+    The caver-js library is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    The caver-js library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+    GNU Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    along with the caver-js. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { Wallet } from '../../..'
+import { Result } from '../../caver-abi/src'
+import { KIP37 } from '../../caver-kct/src'
+import KIP7 from '../../caver-kct/src/kip7'
+import RpcCallToMethod, { TransactionReceipt } from '../../caver-rtm/src'
+import { AbiItem } from '../../caver-utils/src'
+import KeyringContainer from '../../caver-wallet/src'
+
+export interface ContractOptions {
+    address: string
+    jsonInterface: AbiItem[]
+    from: string
+    feePayer: string
+    feeDelegation: boolean
+    feeRatio: number | string
+    gasPrice: number | string
+    gas: number | string
+    data: string
+}
+
+export interface ContractDeployParams {
+    data?: string
+    arguments?: any[]
+}
+
+export interface SendOptions {
+    from?: string
+    feeDelegation?: boolean
+    feePayer?: string
+    feeRatio?: number
+    gas?: number | string
+}
+
+export interface SendOptionsWithFormatter extends SendOptions {
+    contractDeployFormatter?: Function
+}
+
+export interface CallOptions {
+    from?: string
+    gasPrice?: string
+    gas?: number
+}
+
+export interface ContractMethod {
+    arguments?: any[]
+    send?: Contract['send']
+    sign?: Contract['sign']
+    signAsFeePayer?: Contract['signAsFeePayer']
+    estimateGas?: RpcCallToMethod['klay_estimateGas']
+    encodeABI?: () => string
+}
+
+export interface EventOptions {
+    filter?: object
+    fromBlock?: number
+    toBlock?: number
+    topics?: string[]
+}
+
+export interface EventData {
+    address?: string
+    blockNumber?: number
+    transactionHash?: string
+    transactionIndex?: number
+    blockHash?: string
+    logIndex?: number
+    id?: string
+    returnValues?: {
+        operator?: string
+        from?: string
+        to?: string
+        id?: string
+        ids?: string[]
+        value?: string
+        values?: string[]
+        account?: string
+        approved?: boolean
+        tokenId?: string
+    }
```

Comment:
May I know where you referenced the spec for `returnValues` type?
In the execution result, record type is returned.
So I think it should be changed to:
```suggestion
export interface EventData<
  ReturnValues extends Record<string, string> = Record<string, string>
> {
  address?: string;
  blockNumber?: number;
  transactionHash?: string;
  transactionIndex?: number;
  blockHash?: string;
  logIndex?: number;
  id?: string;
  returnValues?: ReturnValues;
```

Reply:
Sorry for the late reply. I'm really busy these days 😂

It written based on the test files like `test/packages/caver.kct.kip37.js`.

As you said, it returned depending on the event triggered by the contract.
However, `ReturnValues extends Record<string, string> = Record<string, string>` will not handle interface and object with a value not a string.
Therefore, i suggest `ReturnValues extends Record<keyof ReturnValues, unknown> = Record<string, string>`